notification/context.go

```diff
@@ -0,0 +1,47 @@
+package notification
+
+import (
+	"github.com/flanksource/duty/models"
+	"github.com/flanksource/incident-commander/api"
+	"github.com/flanksource/incident-commander/db"
+	"github.com/google/uuid"
+)
+
+type Context struct {
+	*api.Context
+	notificationID uuid.UUID
+	log            *models.NotificationSendHistory
+}
+
+func NewContext(ctx *api.Context, notificationID uuid.UUID) *Context {
+	return &Context{
+		Context:        ctx,
+		notificationID: notificationID,
+		log:            models.NewNotificationSendHistory(notificationID),
+	}
+}
+
+func (t *Context) StartLog() error {
+	return db.PersistNotificationSendHistory(t.Context, t.log)
+}
+
+func (t *Context) EndLog() error {
+	return db.PersistNotificationSendHistory(t.Context, t.log.End())
+}
+
```

Comment:
We should probably just be called t.db.Save directly, no need for an intermediate function